src/script/sign.cpp

```diff
@@ -19,15 +19,17 @@ using namespace std;
 
 typedef std::vector<unsigned char> valtype;
 
-TransactionSignatureCreator::TransactionSignatureCreator(const CKeyStore* keystoreIn, const CTransaction* txToIn, unsigned int nInIn, const CAmount &amountIn, uint32_t nHashTypeIn) : BaseSignatureCreator(keystoreIn), txTo(txToIn), nIn(nInIn), amount(amountIn), nHashType(nHashTypeIn), checker(txTo, nIn, amount) {}
+TransactionSignatureCreator::TransactionSignatureCreator(const CKeyStore* keystoreIn, const CTransaction* txToIn, unsigned int nInIn, const CAmount &amountIn, uint32_t nHashTypeIn) : BaseSignatureCreator(keystoreIn), txTo(txToIn), nIn(nInIn), amount(amountIn), nHashType(nHashTypeIn), checker(txTo, nIn, amount, (nHashTypeIn & SIGHASH_FORKID) ? SCRIPT_ENABLE_SIGHASH_FORKID:0) {}
 
 bool TransactionSignatureCreator::CreateSig(std::vector<unsigned char>& vchSig, const CKeyID& address, const CScript& scriptCode) const
 {
     CKey key;
     if (!keystore->GetKey(address, key))
         return false;
 
-    uint256 hash = SignatureHash(scriptCode, *txTo, nIn, nHashType, amount);
+    uint256 hash;
+    if (nHashType & SIGHASH_FORKID) hash = SignatureHashBitcoinCash(scriptCode, *txTo, nIn, nHashType, amount);
+    else hash = SignatureHash(scriptCode, *txTo, nIn, nHashType, amount);
```

Comment:
nit: if statement formatting

or alternatively what about 

```c++
hash = SignatureHash(scriptCode, *txTo, nIn, nHashType, amount);
if (nHashType & SIGHASH_FORKID) 
       hash = SignatureHashBitcoinCash(scriptCode, *txTo, nIn, nHashType, amount);
```

or use the same style you apply to https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/740/files#diff-a0ec10dadca1c87cc43ef3113574a264L317